crypto/asn1/a_i2d_fp.c

```diff
@@ -61,6 +61,43 @@
 #include <openssl/mem.h>
 
 
+int ASN1_i2d_bio(i2d_of_void *i2d, BIO *out, void *in) {
+  if (i2d == NULL || out == NULL || in == NULL) {
+    OPENSSL_PUT_ERROR(ASN1, ERR_R_PASSED_NULL_PARAMETER);
+    return 0;
+  }
+
+  int size = i2d(in, NULL);
+  if (size <= 0) {
+    return 0;
+  }
+
+  char *buffer = (char *)OPENSSL_malloc(size);
```

Comment:
Why not use `unsigned char *` as the type for `buffer` and avoid the need for the subsequent cast?